### PR TITLE
Change DistributedMv writeType and deleteSource

### PR DESCRIPTION
**Pull request by bradyoo**
1. When deleting the source, delete it unchecked because if the source is not fully persisted, move should still be successful.
2. When performing a move operation, change the writeType to THROUGH if the source is persisted to not make files unpersisted.

### CODE REVIEW

**Comment by bradyoo** [comment]
@gpang

**Comment by bradyoo** [comment]
Tests in error: 
alluxio.client.cli.fs.command.PersistCommandTest.persistPartial(alluxio.client.cli.fs.command.PersistCommandTest)
  Run 1: PersistCommandTest.persistPartial:297->AbstractFileSystemShellTest.checkFilePersisted:307 » Timeout
  Run 2: PersistCommandTest.persistPartial:297->AbstractFileSystemShellTest.checkFilePersisted:307 » Timeout

alluxio.client.fs.FreeAndDeleteIntegrationTest.deleteDir(alluxio.client.fs.FreeAndDeleteIntegrationTest)
  Run 1: FreeAndDeleteIntegrationTest.deleteDir:146 » TestTimedOut test timed out after...
  Run 2: FreeAndDeleteIntegrationTest.deleteDir:148 » TestTimedOut test timed out after...

flaky.

**Comment by bradyoo** [comment]
alluxio-bot, test this please

**Comment by bradyoo** [comment]
@gpang: Can you take a look at this?

**Comment by bradyoo** [comment]
Ping @gpang

**Comment by bradyoo** [comment]
Tests in error: 
alluxio.server.ft.ZookeeperFailureIntegrationTest.zkFailure(alluxio.server.ft.ZookeeperFailureIntegrationTest)
  Run 1: ZookeeperFailureIntegrationTest.zkFailure:80 » Timeout Timed out waiting for a...
  Run 2: ZookeeperFailureIntegrationTest.zkFailure:80 » Timeout Timed out waiting for a...

**Comment by bradyoo** [comment]
alluxio-bot, test this please.

File: job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java [hunk 1]
```diff
@@ -339,6 +339,14 @@ private static void migrate(MigrateCommand command, WritePType writeType, boolea
     String destination = command.getDestination();
     LOG.debug("Migrating {} to {}", source, destination);
 
+    // If the write type is async through and the source is being deleted but source is persisted,
+    // keep the destination to be persisted by changing its write type
+    if (deleteSource && writeType.equals(WritePType.ASYNC_THROUGH)) {
+      if (fileSystem.getStatus(new AlluxioURI(source)).isPersisted()) {
+        writeType = WritePType.THROUGH;
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java]: This means it will NOT be CACHE. Is that what we want?

Comment (bradyoo) [reply job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java]: I don't actually know. Do you prefer CACHE_THROUGH here?

Comment (bradyoo) [reply job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java]: Ping.

**Comment by bradyoo** [comment]
@gpang: Changed to CACHE_THROUGH given what I think your feedback was.

File: job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java [hunk 2]
```diff
@@ -357,7 +365,8 @@ private static void migrate(MigrateCommand command, WritePType writeType, boolea
       }
     }
     if (deleteSource) {
-      fileSystem.delete(new AlluxioURI(source));
+      fileSystem.delete(new AlluxioURI(source),
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java]: add a comment here why we want an unchecked delete.

**Comment by bradyoo** [comment]
added comment

**Comment by bradyoo** [comment]
I can't merge. Can you merge it for me?

**Comment by gpang** [comment]
I will wait for all the checks

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**